pkg/service/backup/restore_worker_data.go

```diff
@@ -210,10 +211,20 @@ func (w *restoreWorker) workFunc(ctx context.Context, run *RestoreRun, target Re
 	// for creating and downloading batches.
 	// Goroutine returns only in case of error or
 	// if the whole table has been restored.
-	return parallel.Run(len(w.hosts), target.Parallel, func(n int) error {
+	return parallel.Run(len(w.hosts), target.Parallel, func(n int) (err error) {
 		// Current goroutine's host
 		h := &w.hosts[n]
+		defer func() {
+			if err != nil {
+				w.metrics.SetRestoreState(run.ClusterID, w.location, target.SnapshotTag, h.Host, metrics.RestoreStateError)
+				return
+			}
+			w.metrics.SetRestoreState(run.ClusterID, w.location, target.SnapshotTag, h.Host, metrics.RestoreStateIdle)
+		}()
+
 		for {
+			w.metrics.SetRestoreState(run.ClusterID, w.location, target.SnapshotTag, h.Host, metrics.RestoreStateDownloading)
+
 			pr, err := w.prepareRunProgress(ctx, run, target, h, dstDir, srcDir)
```

Comment:
The downloading state should be set only after `prepareRunProgress`, because host might wait in `chooseIDsForBatch` before downloading remaining batches. This also means that the `idle` state should be set at the end of the for loop.

Reply:
Makes sense.
Thanks !